chemprop/cli/hpopt.py

```diff
@@ -0,0 +1,383 @@
+from chemprop.nn.utils import Activation
+from chemprop.nn import AggregationRegistry
+from chemprop.data import MolGraphDataLoader
+from chemprop.cli.utils.command import Subcommand
+from chemprop.cli.train import (
+    add_train_args,
+    build_datasets,
+    build_model,
+    build_splits,
+    normalize_inputs,
+    process_train_args,
+    validate_train_args,
+)
+from chemprop.cli.common import add_common_args, process_common_args, validate_common_args
+import json
+import logging
+import sys
+from argparse import ArgumentParser, Namespace
+from pathlib import Path
+from copy import deepcopy
+
+import torch
+from lightning import pytorch as pl
+from ray import tune
+from ray.train import CheckpointConfig, RunConfig, ScalingConfig
+from ray.train.lightning import (
+    RayDDPStrategy,
+    RayLightningEnvironment,
+    RayTrainReportCallback,
+    prepare_trainer,
+)
+from ray.train.torch import TorchTrainer
+from ray.tune.schedulers import ASHAScheduler
+
+NO_HYPEROPT = False
+try:
+    from ray.tune.search.hyperopt import HyperOptSearch
+except ImportError:
+    NO_HYPEROPT = True
```

Comment:
```suggestion
NO_HYPEROPT = False
try:
    from ray.tune.search.hyperopt import HyperOptSearch
    from ray import tune
    from ray.train import CheckpointConfig, RunConfig, ScalingConfig
    from ray.train.lightning import (
        RayDDPStrategy,
        RayLightningEnvironment,
        RayTrainReportCallback,
        prepare_trainer,
    )
    from ray.train.torch import TorchTrainer
    from ray.tune.schedulers import ASHAScheduler
except ImportError:
    NO_HYPEROPT = True
```


resolves latest test failure here: https://github.com/chemprop/chemprop/actions/runs/8664591034/job/23761373381?pr=734#step:6:136